### PR TITLE
Added utime and sysFsUtime

### PR DESCRIPTION
**Pull request by crystalct**

### CODE REVIEW

**Comment by bucanero** [comment]
@miigotu sure, I'll take a look on the PR, and the syscalls.

about Cygwin, I don't have a windows environment so I can't really check those changes. Is there any Win-dev around?

@crystalct , do you have more Cygwin changes to submit? if you have, perhaps it will be better to create a brach like `cygwin-patch`, and commit all the changes there, and keep it in a dedicated PR.

**Comment by miigotu** [comment]
Yes, please keep changes separated. 1 for adding the syscall, another for changes like cygwin/Mac support.

You don't need to create a new pr to do it, you can force push to the branch you made this PR with.

File: ppu/crt/crt1.c [hunk 1]
```diff
@@ -87,7 +89,8 @@ static void __syscalls_init(void)
 	__syscalls.rmdir_r = __librt_rmdir_r;
 	__syscalls.link_r = __librt_link_r;
 	__syscalls.unlink_r = __librt_unlink_r;
-	__syscalls.access_r = __librt_access_r;
+	//__syscalls.access_r = __librt_access_r;
```
Comment (bucanero) [inline ppu/crt/crt1.c:92]: please uncomment the `access` syscall

File: ppu/librt/utime.c [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+#include <sys/reent.h>
+#include <sys/errno.h>
+#include <sys/types.h>
+#include <sys/lv2errno.h>
+
+#include <sys/file.h>
+#include <utime.h>
+
+
+int _utime(const char *path, const struct utimbuf *times, int* _errno)
```
Comment (bucanero) [inline ppu/librt/utime.c:14]: I'd remove this function, and unify the code in the next method.

File: ppu/librt/utime.c [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+		time_t     now;
+		now = time(NULL);
+		now_time.actime = now;
+		now_time.modtime = now;
+		return lv2errno(sysLv2FsUtime(path,(sysFSUtimbuf *)&now_time));
+	}
+	else return lv2errno(sysLv2FsUtime(path,(sysFSUtimbuf *)times));
+}
+
+int
```
Comment (bucanero) [inline ppu/librt/utime.c:27]: this is my proposed unified function:

```c
int
_DEFUN(__librt_utime_r,(r,path,times),
	   struct _reent *r _AND
	   const char *path _AND
	   const struct utimbuf *times)
{
	sysFSUtimbuf t;
	if (times)
	{
		t.actime = times->actime;
		t.modtime = times->modtime;
	}
	else
	{
		time_t now = time(NULL);
		t.actime = now;
		t.modtime = now;
	}

	return lv2errno_r(r,sysLv2FsUtime(path,&t));
}
```

File: ppu/librt/utime.c [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+_DEFUN(__librt_utime_r,(r,path,times),
+	   struct _reent *r _AND
+	   const char *path _AND
+	   const struct utimbuf *times)
+{
+	return _utime(path,times,&r->_errno);
```
Comment (bucanero) [inline ppu/librt/utime.c:33]: note that your current function _utime() is not updating the `reent` structure `r`. (_errno is never updated)

File: tools/cgcomp/source/main.cpp [hunk 5]
```diff
@@ -4,10 +4,19 @@
...
+	char currdir[1024];
+	char destfile[1026];
+#endif
+
 #if !defined(WIN32)
 #include <dlfcn.h>
 #endif
 
+
+
```
Comment (bucanero) [inline tools/cgcomp/source/main.cpp:19]: I'd try to keep from adding blank lines unless is really needed.

File: ppu/crt/crt1.c (hunk 1 above)
Comment (bucanero) [reply ppu/crt/crt1.c:92]: also, at least on my local build, the `#include <utime.h>` is not needed

File: ppu/crt/crt1.c [hunk 6]
```diff
@@ -4,6 +4,7 @@
 #include <sys/time.h>
 #include <sys/dirent.h>
 #include <sys/syscalls.h>
+#include <utime.h>
```
Comment (bucanero) [inline ppu/crt/crt1.c:7]: at least on my local build, the `#include <utime.h>` is not needed

**Comment by miigotu** [comment]
@crystalct please don't get discouraged and give up lol, we appreciate your contributions!

**Comment by crystalct** [comment]
It was my first pull request :P
Next will be better  ^_^
About cygwin, i work with it but 64bit version. After ps3 toolchain, psl1ght and ps3libraries installation, cgcomp does't work. Source code was correct so it was hard to understand the problem.
Cgcomp + cg.dll (64 bit) compiled source shaders using relative path but not full path sources.
That is a workaround. I should try to operate at 32bit (all intsallations of course) to see if the problem is the same.

**Comment by crystalct** [comment]
Branch created for cygwin.
Utime modified as Bucanero tips.

**Comment by crystalct** [comment]
About cygwin.. i confirm 32bit version has the same problem. In addition i discover that gcc inside a cygwin installation doesn't have WIN32 defined,bacause it's POSIX. So if in sources code is necessary a #ifdef WIN32, with cygwin it's not true.

**Comment by crystalct** [comment]
@bucanero, access still old not working version.

**Comment by bucanero** [comment]
> @bucanero, access still old not working version.

Right, I've just submitted a fix based on your feedback #86  👍

**Pull request merged**